### PR TITLE
Add drag from clipboard to journal to store data

### PR DESCRIPTION
**Pull request by erilyth**
There was a problem with the old PR, for some reason it showed that I wanted to merge from "unknown repository", so my updates weren't being reflected there.

@samdroid-apps Sorry for taking so long, I had exams till day before.
I fixed the problem with the pixbufs. This was happening mainly with the bmp images, but it works now.

Old PR - https://github.com/sugarlabs/sugar/pull/639

### CODE REVIEW

**Comment by erilyth** [comment]
@samdroid-apps any updates on this? Did you find a bug?

File: src/jarabe/frame/activitiestray.py [hunk 1]
```diff
@@ -90,6 +107,49 @@ def __init__(self, home_activity, group):
...
+        self._context_map.add_context(context, 0, len(context_targets))
+
+        self._cb_object = ClipboardObject(0, "")
+
+        for target in context_targets:
+            if str(target) not in ('TIMESTAMP', 'TARGETS', 'MULTIPLE'):
+                widget.drag_get_data(context, target, time)
+
+        Gdk.drop_finish(context, True, Gtk.get_current_event_time())
+        if hasattr(self, "_cb_object") and self._cb_object is not None:
```
Comment (samdroid-apps) [inline src/jarabe/frame/activitiestray.py]: do you need the hasattr checks?  You set it to None at the top right?

File: src/jarabe/frame/activitiestray.py [hunk 2]
```diff
@@ -90,6 +107,49 @@ def __init__(self, home_activity, group):
...
+        if selection.get_pixbuf() is not None:
+            data = selection.get_pixbuf()
+        if len(selection.get_uris()) != 0:
+            data = selection.get_uris()[0]
+        if selection.get_text() is not None:
+            data = selection.get_text()
+        if data is None:
+            data = selection.get_data()
+
+        if (data is not None and hasattr(self, "_cb_object") and
```
Comment (samdroid-apps) [inline src/jarabe/frame/activitiestray.py]: hasattr check again :(

**Comment by samdroid-apps** [comment]
It looks really good.

It seems to have some weird conditions with drags from browse.  If I drag an image from browse to the journal, it fails [1].  But if I go browse->clipboard then clipboard->browse it works.

Also, using BMP is interesting, I'm interested to hear the rational behind this.  Isn't BMP pretty bad on compression?

But wow, I'm so sorry that you're working on this.  Gtk drag n drop looks pretty horrid :(

**Comment by i5o** [comment]
"any updates on this? "

**Comment by erilyth** [comment]
@i5o Sorry I totally forgot about this. I spent a while but I'm kind of stuck on this, I don't really understand why dragging from browse to the journal directly fails but browse->clipboard etc work (as @samdroid-apps mentioned).

**Comment by chimosky** [comment]
Moved to #960.

**Pull request closed without merging**